### PR TITLE
Tidy R pkg

### LINKED ISSUE #700: R: Tidy package
- [x] Apply tidyverse coding style
- [x] Update roxygen version
- [x] Update package description
- [x] Tidy DESCRIPTION

### PR DESCRIPTION
**Pull request by pat-s**
fixes #700 

And adds a `.gitignore` file for the subdir.

This is just a first stab, there is more cleanup to do to get the package to the CRAN level :)

### CODE REVIEW

File: tools/rpkg/NAMESPACE [hunk 1]
```diff
@@ -2,12 +2,7 @@
 
 export(duckdb)
 export(duckdb_shutdown)
-export(duckdb_register)
```
Comment (krlmlr) [inline tools/rpkg/NAMESPACE:5]: Is this intended?

**Review by krlmlr — APPROVED** [review]

File: tools/rpkg/NAMESPACE (hunk 1 above)
Comment (pat-s) [reply tools/rpkg/NAMESPACE:5]: Done automatically by `devtools::document()`. We'll see along the way why this happened/what was missing when focusing more on documentation.

Comment (krlmlr) [reply tools/rpkg/NAMESPACE:5]: @hannesmuehleisen Did you mean to export `duckdb_register()` and friends?

**Review by hannes — APPROVED** [review]
Thanks

**Comment by hannesmuehleisen** [comment]
Need to figure out why Travis is unhappy still

**Comment by hannesmuehleisen** [comment]
Ah: The package does no longer export the `duckdb_startup_R` function:
`error in evaluating the argument 'drv' in selecting a method for function 'dbConnect': object 'duckdb_startup_R' not found`
Probably because i hand-edited the `NAMESPACE` file.

**Comment by pat-s** [comment]
> Probably because i hand-edited the NAMESPACE file.

Most likely the `@export` is missing in the R script for that function and hence `devtools::document()` does not account for it then.

I'll fix it after lunch so that Travis passes.

NAMESPACE should usually not be touched manually, `devtools::document()` is our friend here :)

**Comment by pat-s** [comment]
I've automated the registration of the C++ functions via {Rcpp}.
Not sure if this was the plan though, @krlmlr?

If you know how to do this more elegantly / omit {Rcpp}, then please go ahead :)

Also I documented some functions so that they are registered in the NAMEPSPACE file and won't be dropped. CI should pass now?

**Comment by krlmlr** [comment]
I suspect we don't want the Rcpp dependency. Not sure if we can do the same with cpp11?

**Comment by hannesmuehleisen** [comment]
Indeed I would not like to add a dependency for developer convenience.

File: tools/rpkg/NAMESPACE (hunk 1 above)
Comment (hannesmuehleisen) [reply tools/rpkg/NAMESPACE:5]: Yes.

**Comment by pat-s** [comment]
I'll check later if the `devtools` convenience can somehow be used without {Rcpp}.

**Comment by hannesmuehleisen** [comment]
What happened to `read_csv_duckdb` and `duckdb.read.csv`? Any reason they are not exported any more?

**Comment by pat-s** [comment]
They weren't documented so far. I can do so, no problem.

I just was not sure if they are meant to be internal functions or meant to be exported.

**Comment by hannesmuehleisen** [comment]
Please do

**Comment by krlmlr** [comment]
I see `duckdb.read.csv()` is no longer exported, hope that's fine.

**Comment by pat-s** [comment]
> I see duckdb.read.csv() is no longer exported, hope that's fine.

Due to a double assigment "<- XX <-".

If needed, we can add it again? @hannesmuehleisen

**Pull request merged**